cpp/src/arrow/dataset/file_csv.h

```diff
@@ -37,6 +37,11 @@ class ARROW_DS_EXPORT CsvFileFormat : public FileFormat {
  public:
   /// Options affecting the parsing of CSV files
   csv::ParseOptions parse_options = csv::ParseOptions::Defaults();
+  /// Options affecting how CSV files are read.
+  ///
+  /// Note use_threads is ignored (it is always considered false) and block_size
+  /// should be set on CsvFragmentScanOptions.
+  csv::ReadOptions read_options = csv::ReadOptions::Defaults();
```

Comment:
I'm not sure it's reasonable to add `ReadOptions` here. I think only `skip_rows` and `autogenerate_column_names` are meaningful here. Column renaming can be accomplished in projection if desired (obviating `column_names`) and as noted `use_threads` and `block_size` are configured elsewhere.

Reply:
Makes sense. I was going back and forth on it because of the redundancy. I'll inline all relevant fields and xref ReadOptions for documentation.